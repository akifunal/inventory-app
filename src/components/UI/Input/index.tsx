export type InputProps = {
	id?: string;
	value?: string | number;
	type?: string;
	placeholder?: string;
	className?: string;
	disabled?: boolean;
	required?: boolean;
	min?: number;
	max?: number;
	onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
	onClick?: (e: React.MouseEvent<HTMLInputElement>) => void;
};

const Input = (props: InputProps) => {
	return (
		<input
			id={props.id}
			type={props.type || 'text'}
			onChange={props.onChange}
			onClick={props.onClick}
			value={props.value}
			placeholder={props.placeholder}
			min={props.min}
			max={props.max}
			className='block w-full rounded-lg border border-gray-500 bg-gray-50 p-2.5 text-sm text-gray-900 placeholder-gray-700 focus:border-gray-500 focus:ring-gray-500 dark:border-gray-500 dark:bg-gray-700 dark:text-gray-400 dark:placeholder-gray-500'
		/>
	);
};

export default Input;
