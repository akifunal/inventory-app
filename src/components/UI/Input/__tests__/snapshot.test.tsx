import { render } from '@testing-library/react';
import Input from '../index';

describe('Input component', () => {
	it('should render and match the snapshot with default props', () => {
		// Arrange and Act
		const { asFragment } = render(<Input />);

		// Assert
		expect(asFragment()).toMatchSnapshot();
	});

	it('should render with the correct props', () => {
		// Arrange and Act
		const { asFragment } = render(
			<Input id='testid' value={10} type='number' />
		);

		// Assert
		expect(asFragment()).toMatchSnapshot();
	});
});
