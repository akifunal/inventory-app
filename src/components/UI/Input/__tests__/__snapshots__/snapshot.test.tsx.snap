// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render and match the snapshot with default props 1`] = `
<DocumentFragment>
  <input
    class="block w-full rounded-lg border border-gray-500 bg-gray-50 p-2.5 text-sm text-gray-900 placeholder-gray-700 focus:border-gray-500 focus:ring-gray-500 dark:border-gray-500 dark:bg-gray-700 dark:text-gray-400 dark:placeholder-gray-500"
    type="text"
    value=""
  />
</DocumentFragment>
`;

exports[`Input component should render with the correct props 1`] = `
<DocumentFragment>
  <input
    class="block w-full rounded-lg border border-gray-500 bg-gray-50 p-2.5 text-sm text-gray-900 placeholder-gray-700 focus:border-gray-500 focus:ring-gray-500 dark:border-gray-500 dark:bg-gray-700 dark:text-gray-400 dark:placeholder-gray-500"
    id="testid"
    type="number"
    value="10"
  />
</DocumentFragment>
`;
